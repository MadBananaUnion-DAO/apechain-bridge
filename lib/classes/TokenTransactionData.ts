import { immerable } from 'immer';
import { TokenInfo } from '@decent.xyz/box-common';

export class TokenTransactionData {
  [immerable] = true;
  token: TokenInfo;
  amount: string;
  amountUsd: string = '';

  constructor(token: TokenInfo, amount: string = '') {
    this.token = token;
    this.amount = amount;
  }
}
