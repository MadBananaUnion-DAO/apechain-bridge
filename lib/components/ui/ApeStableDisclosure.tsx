import { Address, isAddressEqual } from 'viem';
import {
  ApeEthOmnichainContract,
  ApeUsdOmnichainContract,
} from '../../utils/utils';
import AdaptiveTooltip from './tooltip/AdaptiveTooltip';
import { useMemo } from 'react';
import { InfoIcon } from '../icons/InfoIcon';

type ApeStableDisclosureProps = {
  isSourceToken: boolean;
  tokenAddress: Address;
  tokenUsdValue: string;
};

/**
 * Display a warning icon with tooltip if:
 *  - the token is the destination token, AND
 *  - the token is ApeUSD or ApeETH
 */
export const ApeStableDisclosure = ({
  isSourceToken,
  tokenAddress,
}: ApeStableDisclosureProps) => {
  const isApeUsd = isAddressEqual(tokenAddress, ApeUsdOmnichainContract);
  const isApeEth = isAddressEqual(tokenAddress, ApeEthOmnichainContract);
  const isApeStable = isApeUsd || isApeEth;

  const tooltipContent = useMemo(() => {
    if (isApeEth) {
      return 'The underlying price of APE-ETH is correlated with Liquid Staked ETH (stETH).';
    }
    if (isApeUsd) {
      return 'The underlying price of APE-USD is correlated with DAI.';
    }
  }, [isApeEth, isApeUsd]);

  if (isSourceToken || !isApeStable || !tooltipContent) {
    return;
  }

  return (
    <div className="aw-ml-2 aw-flex aw-items-center">
      <AdaptiveTooltip content={tooltipContent} className="aw-text-white">
        <InfoIcon className="aw-cursor-help" />
      </AdaptiveTooltip>
    </div>
  );
};
