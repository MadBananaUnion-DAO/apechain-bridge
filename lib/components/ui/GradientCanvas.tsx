import { motion } from 'framer-motion';

/**
 * A canvas element that is used to render a Granim gradient.
 *
 * The canvas is delayed by 0.1 seconds to allow the parent component
 * to render first. This prevents the gradient from briefly appearing
 * by itself. The delay is arbitrary and can be adjusted.
 *
 * @param id
 * @constructor
 */
export const GradientCanvas = ({ id }: { id: string }) => {
  return (
    <motion.canvas
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      transition={{ delay: 0.1 }}
      id={id}
      className={'aw-absolute aw-size-full aw-rounded-[10px]'}
    />
  );
};
