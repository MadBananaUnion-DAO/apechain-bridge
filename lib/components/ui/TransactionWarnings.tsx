import { useMemo } from 'react';
import { isAddress } from 'viem';
import { cn, shortenAddress } from '../../utils/utils';
import { BridgeTransactionData } from '../../classes/BridgeTransactionData';

export const HIGH_SLIPPAGE_WARNING =
  'Slippage higher than 1% may be frontrun and result in unfavorable prices.';
const PriceImpactRegex = new RegExp(
  'try sending smaller amounts to reduce price impact.',
  'i',
);

export type TransactionWarningsProps = {
  bridgeTransactionData: BridgeTransactionData;
  destinationAddress?: string | undefined;
  isTopTrader?: boolean;
  className?: string;
};

export const TransactionWarnings = ({
  bridgeTransactionData,
  destinationAddress,
  isTopTrader,
  className,
}: TransactionWarningsProps) => {
  const fundsReceiverWarning = useMemo(() => {
    if (destinationAddress && isAddress(destinationAddress)) {
      return isTopTrader
        ? `Funds will arrive in your Top Trader wallet and will be ready for you to enter Top Trader Tournaments`
        : `Funds will be sent to a different wallet address! ${shortenAddress(destinationAddress)}`;
    }
  }, [destinationAddress, isTopTrader]);

  const warnings = useMemo(() => {
    const warnings = [
      ...(bridgeTransactionData?.priceImpactWarning
        ? [bridgeTransactionData.priceImpactWarning]
        : []),
      ...(bridgeTransactionData?.timeWarning
        ? [bridgeTransactionData.timeWarning]
        : []),
      ...(bridgeTransactionData?.isSlippageHigh ? [HIGH_SLIPPAGE_WARNING] : []),
      ...(fundsReceiverWarning ? [fundsReceiverWarning] : []),
    ];
    return warnings;
  }, [bridgeTransactionData, fundsReceiverWarning]);
  const isMoreThanOneWarning = warnings.length > 1;
  const isTopTraderWarning = warnings.some((w) => w.includes('Top Trader'));
  const showWarningLabel = isMoreThanOneWarning || !isTopTraderWarning;

  if (warnings.length === 0) {
    return;
  }

  const singlePriceImpactWarning =
    bridgeTransactionData?.priceImpactWarning && !isMoreThanOneWarning;

  return (
    <div
      className={cn(
        'aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2',
        className,
        bridgeTransactionData?.priceImpactWarning
          ? 'aw-text-danger'
          : 'aw-text-warning',
      )}
      id={isTopTraderWarning ? 'aw-top-trader-warning' : ''}
    >
      {showWarningLabel && (
        <span className="aw-mr-1.5 aw-font-bold">Warning:</span>
      )}
      {isMoreThanOneWarning ? (
        <ul className="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]">
          {warnings.map((warning) => {
            const priceImpactWarning = PriceImpactRegex.test(warning);
            return (
              <li
                className={cn({
                  'aw-text-danger': priceImpactWarning,
                  'aw-text-warning': !priceImpactWarning,
                })}
                key={warning}
              >
                {warning}
              </li>
            );
          })}
        </ul>
      ) : (
        <span
          className={cn({
            'aw-text-danger': singlePriceImpactWarning,
          })}
        >
          {warnings[0]}
        </span>
      )}
    </div>
  );
};
