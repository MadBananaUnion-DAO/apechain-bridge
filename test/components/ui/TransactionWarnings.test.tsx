import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { TransactionWarnings } from '../../../lib/components/ui/TransactionWarnings';
import { BridgeTransactionData } from '../../../lib/classes/BridgeTransactionData';
import { WARNING_PRICE_IMPACT } from '../../../lib/utils/constants';

const createBridgeTransactionData = (
  timeWarning?: string,
  slippagePercentage?: number,
  priceImpactWarning?: string,
) => {
  const bridgeTransactionData = new BridgeTransactionData();
  bridgeTransactionData.timeWarning = timeWarning;
  bridgeTransactionData.priceImpactWarning = priceImpactWarning;
  bridgeTransactionData.slippagePercentage = slippagePercentage ?? 1;
  return bridgeTransactionData;
};

const exampleTimeWarning = 'The estimated waiting time to bridge is 7 days.';
const examplePriceImpactWarning = WARNING_PRICE_IMPACT(0.023);

describe('components/ui/TransactionWarnings', () => {
  test('should return nothing if there are no warnings', async () => {
    const bridgeTransactionData = createBridgeTransactionData(undefined, 1);
    const transactionWarnings = render(
      <TransactionWarnings bridgeTransactionData={bridgeTransactionData} />,
    );
    // Expect an empty component, which with @testing-library/react.render
    //   seems to be an empty div
    expect(transactionWarnings.container).toMatchInlineSnapshot(`<div />`);
    expect(transactionWarnings).toMatchSnapshot();
  });

  test('should render warnings when isTopTrader is FALSE', async () => {
    const bridgeTransactionData = createBridgeTransactionData(
      exampleTimeWarning,
      1.01,
      examplePriceImpactWarning,
    );
    const destinationAddress = '0x1234567890123456789012345678901234567890';
    const transactionWarnings = render(
      <TransactionWarnings
        bridgeTransactionData={bridgeTransactionData}
        destinationAddress={destinationAddress}
      />,
    );
    const warningText = screen.getByText(exampleTimeWarning);
    const slippageText = screen.getByText(
      /slippage higher than 1% may be frontrun and result in unfavorable prices/i,
    );
    const receiverAddressText = screen.getByText(
      /funds will be sent to a different wallet address! 0x1234...7890/i,
    );
    const priceImpactText = screen.getByText(examplePriceImpactWarning);

    expect(warningText).toBeDefined();
    expect(slippageText).toBeDefined();
    expect(priceImpactText).toBeDefined();
    expect(receiverAddressText).toBeDefined();
    expect(transactionWarnings).toMatchSnapshot();
  });

  test('should render warnings when isTopTrader is TRUE', async () => {
    const bridgeTransactionData = createBridgeTransactionData(
      exampleTimeWarning,
      1.01,
      examplePriceImpactWarning,
    );
    const destinationAddress = '0x1234567890123456789012345678901234567890';
    const transactionWarnings = render(
      <TransactionWarnings
        bridgeTransactionData={bridgeTransactionData}
        destinationAddress={destinationAddress}
        isTopTrader={true}
      />,
    );
    const warningText = screen.getByText(exampleTimeWarning);
    const slippageText = screen.getByText(
      /slippage higher than 1% may be frontrun and result in unfavorable prices/i,
    );
    const receiverAddressText = screen.getByText(
      /funds will arrive in your top trader wallet and will be ready for you to enter top trader tournaments/i,
    );
    const priceImpactText = screen.getByText(examplePriceImpactWarning);

    expect(warningText).toBeDefined();
    expect(slippageText).toBeDefined();
    expect(priceImpactText).toBeDefined();
    expect(receiverAddressText).toBeDefined();
    expect(transactionWarnings).toMatchSnapshot();
  });

  test('should render warnings without slippage', async () => {
    const bridgeTransactionData = createBridgeTransactionData(
      exampleTimeWarning,
      undefined,
      examplePriceImpactWarning,
    );
    const destinationAddress = '0x1234567890123456789012345678901234567890';
    const transactionWarnings = render(
      <TransactionWarnings
        bridgeTransactionData={bridgeTransactionData}
        destinationAddress={destinationAddress}
      />,
    );
    const warningText = screen.getByText(exampleTimeWarning);
    const receiverAddressText = screen.getByText(
      /funds will be sent to a different wallet address! 0x1234...7890/i,
    );
    const priceImpactText = screen.getByText(examplePriceImpactWarning);

    expect(warningText).toBeDefined();
    expect(priceImpactText).toBeDefined();
    expect(receiverAddressText).toBeDefined();
    expect(transactionWarnings).toMatchSnapshot();
  });

  test('should render warnings without receiver address', async () => {
    const bridgeTransactionData = createBridgeTransactionData(
      exampleTimeWarning,
      1.01,
      examplePriceImpactWarning,
    );
    const transactionWarnings = render(
      <TransactionWarnings bridgeTransactionData={bridgeTransactionData} />,
    );
    const warningText = screen.getByText(exampleTimeWarning);
    const slippageText = screen.getByText(
      /slippage higher than 1% may be frontrun and result in unfavorable prices/i,
    );
    const priceImpactText = screen.getByText(examplePriceImpactWarning);

    expect(warningText).toBeDefined();
    expect(slippageText).toBeDefined();
    expect(priceImpactText).toBeDefined();
    expect(transactionWarnings).toMatchSnapshot();
  });

  test('should render price impact warning only', async () => {
    const bridgeTransactionData = createBridgeTransactionData(
      undefined,
      1.01,
      examplePriceImpactWarning,
    );
    const transactionWarnings = render(
      <TransactionWarnings bridgeTransactionData={bridgeTransactionData} />,
    );
    const priceImpactText = screen.getByText(examplePriceImpactWarning);

    expect(priceImpactText).toBeDefined();
    expect(transactionWarnings).toMatchSnapshot();
  });
});
