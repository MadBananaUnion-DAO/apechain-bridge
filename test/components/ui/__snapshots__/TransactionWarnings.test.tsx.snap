// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/ui/TransactionWarnings > should render price impact warning only 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
        id=""
      >
        <span
          class="aw-mr-1.5 aw-font-bold"
        >
          Warning:
        </span>
        <ul
          class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
        >
          <li
            class="aw-text-danger"
          >
            You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
          </li>
          <li
            class="aw-text-warning"
          >
            Slippage higher than 1% may be frontrun and result in unfavorable prices.
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
      id=""
    >
      <span
        class="aw-mr-1.5 aw-font-bold"
      >
        Warning:
      </span>
      <ul
        class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
      >
        <li
          class="aw-text-danger"
        >
          You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
        </li>
        <li
          class="aw-text-warning"
        >
          Slippage higher than 1% may be frontrun and result in unfavorable prices.
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/ui/TransactionWarnings > should render warnings when isTopTrader is FALSE 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
        id=""
      >
        <span
          class="aw-mr-1.5 aw-font-bold"
        >
          Warning:
        </span>
        <ul
          class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
        >
          <li
            class="aw-text-danger"
          >
            You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
          </li>
          <li
            class="aw-text-warning"
          >
            The estimated waiting time to bridge is 7 days.
          </li>
          <li
            class="aw-text-warning"
          >
            Slippage higher than 1% may be frontrun and result in unfavorable prices.
          </li>
          <li
            class="aw-text-warning"
          >
            Funds will be sent to a different wallet address! 0x1234...7890
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
      id=""
    >
      <span
        class="aw-mr-1.5 aw-font-bold"
      >
        Warning:
      </span>
      <ul
        class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
      >
        <li
          class="aw-text-danger"
        >
          You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
        </li>
        <li
          class="aw-text-warning"
        >
          The estimated waiting time to bridge is 7 days.
        </li>
        <li
          class="aw-text-warning"
        >
          Slippage higher than 1% may be frontrun and result in unfavorable prices.
        </li>
        <li
          class="aw-text-warning"
        >
          Funds will be sent to a different wallet address! 0x1234...7890
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/ui/TransactionWarnings > should render warnings when isTopTrader is TRUE 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
        id="aw-top-trader-warning"
      >
        <span
          class="aw-mr-1.5 aw-font-bold"
        >
          Warning:
        </span>
        <ul
          class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
        >
          <li
            class="aw-text-danger"
          >
            You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
          </li>
          <li
            class="aw-text-warning"
          >
            The estimated waiting time to bridge is 7 days.
          </li>
          <li
            class="aw-text-warning"
          >
            Slippage higher than 1% may be frontrun and result in unfavorable prices.
          </li>
          <li
            class="aw-text-warning"
          >
            Funds will arrive in your Top Trader wallet and will be ready for you to enter Top Trader Tournaments
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
      id="aw-top-trader-warning"
    >
      <span
        class="aw-mr-1.5 aw-font-bold"
      >
        Warning:
      </span>
      <ul
        class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
      >
        <li
          class="aw-text-danger"
        >
          You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
        </li>
        <li
          class="aw-text-warning"
        >
          The estimated waiting time to bridge is 7 days.
        </li>
        <li
          class="aw-text-warning"
        >
          Slippage higher than 1% may be frontrun and result in unfavorable prices.
        </li>
        <li
          class="aw-text-warning"
        >
          Funds will arrive in your Top Trader wallet and will be ready for you to enter Top Trader Tournaments
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/ui/TransactionWarnings > should render warnings without receiver address 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
        id=""
      >
        <span
          class="aw-mr-1.5 aw-font-bold"
        >
          Warning:
        </span>
        <ul
          class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
        >
          <li
            class="aw-text-danger"
          >
            You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
          </li>
          <li
            class="aw-text-warning"
          >
            The estimated waiting time to bridge is 7 days.
          </li>
          <li
            class="aw-text-warning"
          >
            Slippage higher than 1% may be frontrun and result in unfavorable prices.
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
      id=""
    >
      <span
        class="aw-mr-1.5 aw-font-bold"
      >
        Warning:
      </span>
      <ul
        class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
      >
        <li
          class="aw-text-danger"
        >
          You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
        </li>
        <li
          class="aw-text-warning"
        >
          The estimated waiting time to bridge is 7 days.
        </li>
        <li
          class="aw-text-warning"
        >
          Slippage higher than 1% may be frontrun and result in unfavorable prices.
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/ui/TransactionWarnings > should render warnings without slippage 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
        id=""
      >
        <span
          class="aw-mr-1.5 aw-font-bold"
        >
          Warning:
        </span>
        <ul
          class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
        >
          <li
            class="aw-text-danger"
          >
            You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
          </li>
          <li
            class="aw-text-warning"
          >
            The estimated waiting time to bridge is 7 days.
          </li>
          <li
            class="aw-text-warning"
          >
            Funds will be sent to a different wallet address! 0x1234...7890
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aw-flex aw-flex-col aw-text-left aw-w-full aw-justify-between aw-border-t aw-border-white/10 aw-py-2 aw-text-[14px] aw-leading-[14px] aw-tracking-[0.12px] aw-gap-y-2 aw-text-danger"
      id=""
    >
      <span
        class="aw-mr-1.5 aw-font-bold"
      >
        Warning:
      </span>
      <ul
        class="aw-ml-4 aw-list-disc aw-space-y-2 aw-leading-[18px]"
      >
        <li
          class="aw-text-danger"
        >
          You will receive 2.3% less than the amount sent (in USD equivalent). Try sending smaller amounts to reduce price impact.
        </li>
        <li
          class="aw-text-warning"
        >
          The estimated waiting time to bridge is 7 days.
        </li>
        <li
          class="aw-text-warning"
        >
          Funds will be sent to a different wallet address! 0x1234...7890
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/ui/TransactionWarnings > should return nothing if there are no warnings 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
